app/controllers/stories_controller.rb

```diff
@@ -407,4 +409,12 @@ def verify_user_can_submit_stories
       return redirect_to "/"
     end
   end
+
+  def track_story_reads
+    story = Story.where(short_id: params[:id]).first!
+    @ribbon = ReadRibbon.where(user_id: @user.id, story_id: story.id).first_or_create
+    yield
+    @ribbon.updated_at = Time.now
+    @ribbon.save
```

Comment:
Maybe use `@ribbon.touch` here?

Reply:
Good idiom to use, thanks. I don't know what @hmadison's schedule is this morning, so I'm going to merge and then commit that myself.